openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,263 @@
+        taxa_list = None
+        title = ""
+        if taxa:
+            taxa = [
+                _.get("canonicalName")
+                for _ in taxa
+                if _.get("canonicalName") is not None
+            ]
+            taxa_list = [_.get("string", "") for _ in taxa]
+
+        if taxa_list:
+            title = taxa_list[0]
+
+        return taxa_list, title
+
+    @staticmethod
+    def _get_creator_details(result: dict) -> tuple[str | None, str | None, str | None]:
+        credit_line = None
+        pub_date = None
+        creator = None
+        submitter = result.get("submitter", {})
+        first_name = submitter.get("firstName")
+        last_name = submitter.get("lastName")
+        if first_name and last_name:
+            creator = f"{first_name} {last_name}".strip()
+
+        if credit := result.get("credit"):
+            credit_line = credit.strip()
+            pub_date = result.get("submitted").strip()
+
+        return creator, credit_line, pub_date
+
+    @staticmethod
+    def _get_meta_data(result: dict) -> dict | None:
+        meta_data = {}
+        _uuid = result.get("uid")
+        license_url = result.get("licenseURL")
+
+        img_url, width, height = PhylopicDataIngester._get_image_info(result, _uuid)
+
+        if img_url is None:
+            return None
+
+        meta_data["taxa"], title = PhylopicDataIngester._get_taxa_details(result)
+
+        foreign_url = PhylopicDataIngester._image_url(_uuid)
+
+        (
+            creator,
+            meta_data["credit_line"],
+            meta_data["pub_date"],
+        ) = PhylopicDataIngester._get_creator_details(result)
+
+        return {
+            "foreign_identifier": _uuid,
+            "foreign_landing_url": foreign_url,
+            "image_url": img_url,
+            "license_info": get_license_info(license_url=license_url),
+            "width": width,
+            "height": height,
+            "creator": creator,
+            "title": title,
+            "meta_data": meta_data,
+        }
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        _uuid = data.get("uid")
+        if not _uuid:
+            return
+        logger.debug(f"Processing UUID: {_uuid}")
+        params = {
+            "options": " ".join(
+                [
+                    "credit",
+                    "licenseURL",
+                    "pngFiles",
+                    "submitted",
+                    "submitter",
+                    "taxa",
+                    "canonicalName",
+                    "string",
+                    "firstName",
+                    "lastName",
+                ]
+            )
+        }
+        endpoint = f"{self.base_endpoint}/{_uuid}"
+        response_json = self.get_response_json(params, endpoint)
+        result = self._get_response_data(response_json)
+        if not result:
+            return None
+
+        return self._get_meta_data(result)
```

Comment:
Total nit, but `_get_meta_data` is slightly confusing because `meta_data` is itself just one of the fields being extracted.

Reply:
I've removed this function entirely, it wasn't really serving a useful extra distinction and it made the class method calls easier!